### ISSUE TITLE
Update authorization type for new version home-assistant.

### ISSUE THREAD

**Issue by radroxx** [mueslo/openwrt_hass_devicetracker#4]
You need change X-HA-Access header to "authorization: Bearer" for Long Token.

**Comment by mueslo** [comment]
AFAICT the legacy API password method should still work and I won't be changing it yet since the HA authentication system is still in a state of flux

**Comment by radroxx** [comment]
After the update home-assistant, the utility did not work for me. Maybe I did not fully understand. But you can also add to the config file the ability to choose a method.

**Comment by partofthething** [comment]
@radroxx did you still have the `legacy_api_password` authentication type operational in your hass config when this stopped working for you?

**Comment by radroxx** [comment]
I use the official docker container, after updating the utility stopped working. I stopped receiving notifications from the home-assistant.
All the settings were the same as for the old version of the home-assistant.
A couple of days later I created the first user, disabled support Legacy Api. The user added an api token, and edited the functions.sh on my openwrt after which it all worked. I not use `legacy_api_password` options, maybe this options fix all problem but my variant not bad. I create this issue so that you add support for the new authorization method in your utility. If you do not see a reason to do this, and the  `legacy_api_password` option really works, let's just close this problem.

**Comment by radroxx** [comment]
> @radroxx did you still have the `legacy_api_password` authentication type operational in your hass config when this stopped working for you?

I use the official docker container, after updating the utility stopped working. I stopped receiving notifications from the home-assistant.
All the settings were the same as for the old version of the home-assistant.

**Comment by partofthething** [comment]
I can confirm this. I'm on 0.78 and set the legacy_api_password:

```
  auth_providers:
      - type: homeassistant
      - type: legacy_api_password
```

and it started working again... for now!

**Comment by mueslo** [comment]
> 
> 
> I can confirm this. I'm on 0.78 and set the legacy_api_password:
> 
> ```
>   auth_providers:
>       - type: homeassistant
>       - type: legacy_api_password
> ```
> 
> and it started working again... for now!

Is it really necessary to manually enable legacy_api_password auth? The docs state that all that's necessary is setting the api password in the configuration like so:

    http:
      api_password: yourapipassword

**Comment by rytilahti** [comment]
At the moment `api_password` will automatically enable the legacy api provider. Long-lived access tokens are merely supported since the newest release (https://github.com/home-assistant/home-assistant/pull/16453), so I think the support for the legacy API will stay there for a while. Only problem with that is the log entries related to suggesting conversion to those.

**Comment by ioangogo** [comment]
I might work on implementing support for the long lived tokens that where also introduced in auth update

**Comment by ioangogo** [comment]
I have some code written, just need to package it and test it on my router

**Comment by Cadair** [comment]
Closed by #5 ?
